Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/README.md

```diff
@@ -0,0 +1,298 @@
+| `--api-gateway-emulator-port` | Port for API Gateway | API Gateway Mode |
+| `--api-gateway-emulator-mode` | API Gateway mode (Rest/HttpV1/HttpV2) | API Gateway Mode |
+| `--no-launch-window` | Disable auto-launching web interface | Optional |
+
+
+## API Gateway Configuration
+When using API Gateway mode, you need to configure the route mapping using the APIGATEWAY_EMULATOR_ROUTE_CONFIG environment variable. This can be a single route or an array of routes:
+
+### Single Route Configuration
+```
+{
+    "LambdaResourceName": "AddLambdaFunction",
+    "HttpMethod": "Get",
+    "Path": "/add/{x}/{y}"
+}
+```
+
+### Multiple Routes Configuration
+
+```
+[
+    {
+        "LambdaResourceName": "AddLambdaFunction",
+        "HttpMethod": "Get",
+        "Path": "/add/{x}/{y}"
+    },
+    {
+        "LambdaResourceName": "SubtractLambdaFunction",
+        "HttpMethod": "Get",
+        "Path": "/minus/{x}/{y}"
+    }
+]
+
+```
+
+## Running the Test Tool Project via an IDE
+
+Download this repository and create/modify the launch settings `Properties/launchSettings.json` to startup the test tool
+
+```json
+{
+    "profiles": {
+        "Lambda Test Tool": {
+          "commandName": "Project",
+            "commandLineArgs": "start --lambda-emulator-port 5050 --api-gateway-emulator-port 5051 --api-gateway-emulator-mode Rest",
+            "environmentVariables": {
+                "APIGATEWAY_EMULATOR_ROUTE_CONFIG": {
+                    "LambdaResourceName": "AddLambdaFunction",
+                    "HttpMethod": "Get",
+                    "Path": "/add/{x}/{y}"
+                }
+            }
+        }
+    }
+}
+
+```
+
+## Example Lambda Function Setup
+
+Here's a simple Lambda function that adds two numbers together. This can be implemented in two ways:
+
+### 1. Lambda Function Code
+Here's a simple Lambda function that adds two numbers together:
+
+#### Option 1: Using Top-Level Statements
+
+
+```csharp
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.RuntimeSupport;
+using Amazon.Lambda.Serialization.SystemTextJson;
+
+var Add = (APIGatewayHttpApiV2ProxyRequest request, ILambdaContext context) =>
+{
+    // Parse x and y from the path parameters
+    var x = int.Parse(request.PathParameters["x"]);
+    var y = int.Parse(request.PathParameters["y"]);
+    return (x + y).ToString();
+};
+
+await LambdaBootstrapBuilder.Create(Add, new CamelCaseLambdaJsonSerializer())
+    .Build()
+    .RunAsync();
+
+```
+
+#### Option 2: Using Class Library
+```
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+
+namespace MyLambdaFunction;
+
+public class Function
+{
+    public int Add(APIGatewayHttpApiV2ProxyRequest request, ILambdaContext context)
+    {
+        var x = int.Parse(request.PathParameters["x"]);
+        var y = int.Parse(request.PathParameters["y"]);
+        return x + y;
+    }
+}
+```
+
+### 2. Configuration Files
```

Comment:
I would organize this different and out the information about editing the launchsettings.json file under each of the options above. That way you can say for executable set the launchsetting to this. For class library set the launchsettings to this version. And be sure to explain the  `AWS_LAMBDA_RUNTIME_API` environment variable.

Reply:
i have updated this section and also fixed the launchSettings for the classlibrary and have also added a snippet explaining the AWS_LAMBDA_RUNTIME_API  variable